### ISSUE TITLE
A problem during inference on my own datasets

### ISSUE THREAD

**Issue by stellating** [tianzhi0549/FCOS#133]
I have trained FCOS on my own dataset which has 3 classes, and I'v got a model with 87500 iterations. But when I try to take an inference on my own datasets, I get the following massage:

Running per image evaluation...
Evaluate annotation type *bbox*
DONE (t=0.62s).
Accumulating evaluation results...
DONE (t=0.22s).
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = -1.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = -1.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = -1.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = -1.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.000
Maximum f-measures for classes:
[-0.0, -0.0, 0.0]
Score thresholds for classes (used in demos for visualization purposes):
[-1.0, -1.0, 0.9576103687286377]
2019-09-06 10:20:00,552 fcos_core.inference INFO: OrderedDict([('bbox', OrderedDict([('AP', 0.0), ('AP50', 0.0), ('AP75', 0.0), ('APs', -1.0), ('APm', -1.0), ('APl', 0.0)]))])

Could you please help me?

**Comment by tianzhi0549** [comment]
@stellating I suggest you visualize your detection results to ensure your model works well.

**Comment by dreamhighchina** [comment]
@stellating the same with you，how do you deal with it？

**Comment by Finniu** [comment]
same as me

**Comment by liudanyang** [comment]
> same as me

I got the same problem. Did you fix it?

**Comment by Finniu** [comment]
> > same as me
> 
> I got the same problem. Did you fix it?

no, I think it is the problem that the num of training class is 81 in source code, but mine is different, and he didn't say how to change it. When i was training, i just load all backbone weights, but still doesn't work, XD. So i decide to build a trainer by myself

**Comment by liudanyang** [comment]
> no, I think it is the problem that the num of training class is 81 in source code, but mine is different, and he didn't say how to change it. When i was training, i just loaded all backbone weights, but still doesn't work, XD. So i decide to build a trainer by myself

Well, you can change the category number in fcos_core/config/default.py. But...I still didn't figure out why this issue happened.

**Comment by liudanyang** [comment]
oh...I found the problem..the fetch_data function that I wrote has a bug....

**Comment by JerryIndus** [comment]
@liudanyang 
I met similar problem, and could you tell me the how wrong about fetch_data function detailly? Maybe, i can learn from it, thank you.

**Comment by punknownq** [comment]
@JerryIndus 
Hi.Have you solved the problem?

**Comment by punknownq** [comment]
@liudanyang Could you tell me how do you fix the problem?

**Comment by pp00704831** [comment]
Hello,
Does anyone fix it?

**Comment by SayHelloToEveryone** [comment]
Does anyone fix it?
